packages/mds-ingest-service/service/event_validators.ts

```diff
@@ -141,7 +150,42 @@ export const { validate: validateGetVehicleEventsFilterParams } = SchemaValidato
     geography_ids: { type: 'array', items: uuidSchema, nullable: true }
   },
   required: ['grouping_type']
-})
+}
+
+const filterColumns = <const>{
+  type: 'object',
+  nullable: true,
+  properties: {
+    event: nullableArray([...new Set(EventColumns)]),
+    telemetry: {
+      type: 'object',
+      properties: {
+        base: nullableArray([...new Set(TelemetryBaseColumns)]),
+        gps: nullableArray([...new Set(TelemetryGpsColumns)])
+      },
+      required: ['gps']
+    },
+    annotation: nullableArray([...new Set(EventAnnotationColumns)])
+  },
+  required: ['event', 'telemetry']
+}
+
+const withColumnsGetVehicleEventsFilterParamsSchema = () => {
+  const { properties, required, ...rest } = noColumnsGetVehicleEventsFilterParamsSchema
+  const domainSchema = {
+    ...rest,
+    properties: { ...properties, columns: filterColumns },
+    required: [...required, 'columns']
+  }
+  return domainSchema as Schema<WithColumnsGetVehicleEventsFilterParams>
```

Comment:
Try throwing in a few more `<const>`'s in front of your schemas (e.g. `<const>{foo: 'bar'}`), you should not need to cast the schema ever.